File: 1468-CheckIfNAndItsDoubleExist/1468-CheckIfNAndItsDoubleExist.cs
// Last updated: 3/26/2026, 1:49:21 AM
public class Solution {
    public bool CheckIfExist(int[] arr) {
        HashSet<int> hs = new HashSet<int>();
        
        foreach(int num in arr)
        {
            if(hs.Contains(2*num) || (num%2==0 && hs.Contains(num/2)))
                return true;
            else
                hs.Add(num);
        }
        
        return false;
    }
}

File: 1468-CheckIfNAndItsDoubleExist/1468-CheckIfNAndItsDoubleExist.java
// Last updated: 3/26/2026, 1:49:24 AM
class Solution {
    public boolean checkIfExist(int[] arr) {
        Set<Integer> set = new HashSet<>();
        
        for(int i=0;i<arr.length;i++){
            if((arr[i]%2==0 && set.contains(arr[i]/2)) || set.contains(arr[i]*2)){
                System.out.println(arr[i]);
                return true;
            }
                
            set.add(arr[i]);
        }
        return false;
    }
}